Signal/src/ViewControllers/ConversationView/ConversationViewController.m

```diff
@@ -1586,23 +1580,21 @@ - (void)updateMessageMappingRangeOptions:(MessagesRangeSizeMode)mode
             // We'd like to include at least N seen messages,
             // to give the user the context of where they left off the conversation.
             const NSUInteger kPreferredSeenMessageCount = 1;
-            targetLength = unreadIndicatorPosition + kPreferredSeenMessageCount;
-        } else {
-            // Default to a single page of messages.
-            targetLength = kYapDatabasePageSize;
+            rangeLength = unreadIndicatorPosition + kPreferredSeenMessageCount;
         }
     }
 
-    // The "page-based" range length may have been increased by loading "prev" pages at the
-    // top of the window.
-    NSUInteger rangeLength;
-    while (YES) {
-        rangeLength = kYapDatabasePageSize * (self.page + 1);
-        if (rangeLength >= targetLength) {
-            break;
-        }
-        self.page = self.page + 1;
-    }
+    // Always try to load at least a single page of messages.
+    rangeLength = MAX(rangeLength, kYapDatabasePageSize);
```

Comment:
I feel like this method is now simpler and more readable.

Reply:
👏 👏 👏 I agree.